### PR TITLE
Next.js basics with payroll system

### LINKED ISSUE #7549: Next.js basics with payroll system
#### Proposed title of the article
Next.js basics with payroll system

#### Proposed article introduction
React.js has very cool features and frameworks and we are going to focus on Next.js, the coolest thing about Next.js is that the back-end is already set for us and the only thing you need to do is connect it to the database and you have a full-stack web application.

The other cool thing that most people don't want to try is that you can use arithmetic operators just like in javascript functions and build an amazing software for daily unavoidable calculations.
We will be creating an amazing basic payroll system in Next.js.

### Table of Content
- [Prerequisites](#Prerequisites)
- [Objectives](#Objectives)
- [Next.js Environment Setup](#Next.js-Environment-Setup)
- [Introduction to Payroll System](Introduction-to-Payroll-System)
- [Payroll Arithmetic Calculations](#payroll-Arithmetic-Calculations)
- [Creating a JSON File and Functions in Next.js](#Creating-a-JSON-File-and-Functions-in-Next.js)
- [Fetching Payroll Data and Computing Calculations in Next.js](#Fetching-Payroll-Data-and-Computing-Calculations-in-Next.js)
- [Conclusion](#Conclusion)

### Conclusion
This tutorial has given us a foundation for doing arithmetic operations in Next.js and an introduction to the payroll system.
Handling payroll involves sending out payslips to employees and a styled view of an application.
In our next tutorial we are going to cover all this and maybe send payments to our employees.

Happy coding!

### PR DESCRIPTION
**Pull request by unholydisaster**
### Article checklist
[Topic approved](https://github.com/section-engineering-education/engineering-education/issues/7549)

This closes #7549

### CODE REVIEW

**Comment by srishilesh** [comment]
@unholydisaster 

Can you please link the issue and your previous PR here?

Will start my tech review soon

**Comment by unholydisaster** [comment]
[Link issue](https://github.com/section-engineering-education/engineering-education/issues/7549)
[Previous PR](https://github.com/section-engineering-education/engineering-education/pull/7531#event-6450192575)

**Comment by louisefindlay23** [comment]
Plagiarism check complete

**Comment by srishilesh** [comment]
Hey @unholydisaster ,

Just checking if i can reopen this PR and start the review?

**Comment by unholydisaster** [comment]
You can reopen

File: content/articles/next.js-basics-with-payroll-system/index.md [hunk 1]
```diff
@@ -0,0 +1,260 @@
+# Next.js basics with payroll system
+
+React.js has very cool features and frameworks and we are going to focus on Next.js, the coolest thing about Next.js is that the back-end is already set for us and the only thing you need to do is connect it to the database and you have a full-stack web application.
+The other cool thing that most people don't want to try is that you can use arithmetic operators just like in javascript functions and build an amazing software for daily unavoidable calculations.
+We will be creating an amazing basic payroll system in Next.js.
```
Comment (srishilesh) [inline content/articles/next.js-basics-with-payroll-system/index.md:5]: A more briefer intro about the project would be good.

File: content/articles/next.js-basics-with-payroll-system/index.md [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+- [Conclusion](#Conclusion)
+
+### Prerequisites
+In order to follow along, you will need to have the following:
+1. Basic knowledge of Javascript.
+2. Basic knowledge of React.js and Next.js.
+3. Fully set development environment.
+
+### Objectives
+This tutorial will equip you with the basic arithmetic operators in Next.js and how to effectively use them in functions, for calculations needed in software programming.
```
Comment (srishilesh) [inline content/articles/next.js-basics-with-payroll-system/index.md:24]: You can move this to the intro

File: content/articles/next.js-basics-with-payroll-system/index.md [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+```bash
+npx create-next-app@latest
+```
+The installation will begin and you will be asked to name the application, let's call it payroll.
+Now type this to enter the project folder and start the server.
+
+```bash
+cd payroll
+Npm run dev
```
Comment (srishilesh) [inline content/articles/next.js-basics-with-payroll-system/index.md:39]:
```suggestion
npm run dev
```

File: content/articles/next.js-basics-with-payroll-system/index.md [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+This information is needed for the calculation of personal gross pay and net pay.
+
+### Payroll Arithmetic Calculations
+In calculating the gross pay we need to multiply the wage rates of individual employees by the number of hours worked plus all allowances earned.
+
+ <h4>Gross pay=wages * hours worked + allowances</h6>
+
+The net pay is calculated by deducting all authorized withholdings and pay deductions from the gross pay.
+
+<h4>Net pay=Gross pay - Income tax - Health insurance</h4>
```
Comment (srishilesh) [inline content/articles/next.js-basics-with-payroll-system/index.md:56]: Remove these html tags. Replace them with ` `` `

File: content/articles/next.js-basics-with-payroll-system/index.md [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+                </Link>
+                <h2>${Grosspay}</h2>
+                <h2>${Netpay}</h2>
+                </div>
+                </>
+                 )
+              })}
+    )
+}
+export async function getServerSideProps(context){
```
Comment (srishilesh) [inline content/articles/next.js-basics-with-payroll-system/index.md:225]: Can you please explain more about this function below?

File: content/articles/next.js-basics-with-payroll-system/index.md [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+At this point, we have our gross pay and net pay displayed on the web.
+
+
+### Conclusion
+This tutorial has given us a foundation for doing arithmetic operations in Next.js and an introduction to the payroll system.
+Handling payroll involves sending out payslips to employees and a styled view of an application.
+In our next tutorial we are going to cover all this and maybe send payments to our employees.
+
+Happy coding!
+
```
Comment (srishilesh) [inline content/articles/next.js-basics-with-payroll-system/index.md:257]: Add a GitHub URL that contains the full source code

File: content/articles/next.js-basics-with-payroll-system/index.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+
```
Comment (srishilesh) [inline content/articles/next.js-basics-with-payroll-system/index.md:258]: Add a `Further reading` section that contains reference materials

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @unholydisaster ,

I've requested for a few changes. Please make sure to go through them carefully.

Ping me when you are done.

Thanks.

**Comment by hectorkambow** [comment]
> You can reopen

Why was the PR closed? 

cc @srishilesh  

If many revisions are needed - we should be closing this PR 👍

**Comment by unholydisaster** [comment]
Hey, I have already finished the suggestions but I have a problem with my local repository, I need a backup of the .git/config file of the engineering-education repo.

**Comment by srishilesh** [comment]
> Hey, I have already finished the suggestions but I have a problem with my local repository, I need a backup of the .git/config file of the engineering-education repo.

@unholydisaster 

Are you not able to push the changes? What issue are you facing exactly?

**Comment by unholydisaster** [comment]
I am not able to push the changes

**Comment by srishilesh** [comment]
@unholydisaster 

Can you share a screenshot of the error?

Or you can try opening a new PR.

**Comment by unholydisaster** [comment]
Hey, I have created the new PR.

**Comment by unholydisaster** [comment]
Next.js basics with payroll system #7653

**Comment by unholydisaster** [comment]
Please notify the reviewer of the new PR

**Comment by unholydisaster** [comment]
I don't know how but my systems app data got erased. Am going to make a new
PR with the changes and notify you.

**Pull request closed without merging**